### PR TITLE
Add Stopwatch.GetElapsedTime

### LINKED ISSUE #65858: [API Proposal]: Stopwatch.GetElapsedTime
### Background and motivation

Stopwatch is a class and allocates, but when used for precision timing, you often don't want the overhead of that allocation, and it can be ergonomically inconvenient to try to cache an instance away somewhere.  Developers that want to avoid that allocation end up using Stopwatch.GetTimestamp() directly, at which point they're having to remember what math to do and repeatedly do it for computing a TimeSpan from two timestamps.  We can expose a helper to make that easier.

### API Proposal

```diff
namespace System.Diagnostics
{
    public class Stopwatch
    {
+        public static TimeSpan GetElapsedTime(long startingTimestamp) => GetElapsedTime(startTimestamp, Stopwatch.GetTimestamp());
+        public static TimeSpan GetElapsedTime(long startingTimestamp, long endingTimestamp) => new TimeSpan((endTimestamp - startTimestamp) * s_tickFrequency);
    }
}
```
(based on the discussion in https://github.com/dotnet/runtime/issues/48570#issuecomment-782849544)

### API Usage

```C#
long starting = Stopwatch.GetTimestamp();
...
long ending = Stopwatch.GetTimestamp();
...
TimeSpan time = Stopwatch.GetElapsedTime(starting, ending);
```
or
```C#
long starting = Stopwatch.GetTimestamp();
...
TimeSpan time = Stopwatch.GetElapsedTime(starting);
```

### Alternative Designs

https://github.com/dotnet/runtime/issues/48570

### Risks

_No response_

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/65858

### CODE REVIEW

**Review by MihaZupan — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs [hunk 1]
```diff
@@ -114,6 +114,19 @@ public static long GetTimestamp()
...
+        /// <returns>A <see cref="TimeSpan"/> for the elapsed time between the starting timestamp and the time of this call.</returns>
+        public static TimeSpan GetElapsedTime(long startingTimestamp) =>
+            GetElapsedTime(startingTimestamp, GetTimestamp());
+
+        /// <summary>Gets the elapsed time between two timestamps retrieved using <see cref="GetTimestamp"/>.</summary>
+        /// <param name="startingTimestamp">The timestamp marking the beginning of the time period.</param>
+        /// <param name="endingTimestamp">The timestamp marking the end of the time period.</param>
+        /// <returns>A <see cref="TimeSpan"/> for the elapsed time between the starting and ending timestamps.</returns>
+        public static TimeSpan GetElapsedTime(long startingTimestamp, long endingTimestamp) =>
+            new TimeSpan((long)((endingTimestamp - startingTimestamp) * s_tickFrequency));
```
Comment (am11) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs:128]: An unrelated question about `s_tickFrequency`
On Unix, we use a constant `1000000000` for frequency calculation, while on Windows we call `Kernel32.QueryPerformanceFrequency()`). Would it make sense to align them? e.g. making it constant for Windows as well, or implementing frequency shim on Unix; something like:
```c
int64_t SystemNative_GetPerformanceCounterFrequency()
{
    const int64_t defaultFrequency = 1000000000;

    struct timespec spec;
    if (clock_getres(CLOCK_MONOTONIC, &spec) != 0) return defaultFrequency;

    assert(spec.tv_sec < 1); // resolution in seconds is too high to be correct
    assert(spec.tv_nsec > 0);

    return defaultFrequency / spec.tv_nsec;
}
```

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs [hunk 2]
```diff
@@ -339,12 +339,8 @@ private void AdjustMaxWorkersActive()
                     return;
                 }
 
-
-                long startTime = _currentSampleStartTime;
                 long endTime = Stopwatch.GetTimestamp();
-                long freq = Stopwatch.Frequency;
-
-                double elapsedSeconds = (double)(endTime - startTime) / freq;
+                double elapsedSeconds = Stopwatch.GetElapsedTime(_currentSampleStartTime, endTime).TotalSeconds;
```
Comment (WeihanLi) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs:343]: Maybe we could use `Stopwatch.GetElapsedTime(_currentSampleStartTime).TotalSeconds` here?

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs:128]: It used to call into a native shim. It was changed in https://github.com/dotnet/runtime/pull/43343.

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs:128]: For reference, Unix always returns something in terms of `seconds + nanoseconds`. Where-as Windows returns a measurement in unknown units where you have to independently query a frequency to convert it to `nanoseconds`.

The static APIs on Stopwatch are meant to be minimal overhead and directly return the "raw units" given by the underlying OS functions. This new helper is what allows it to be normalized to `TimeSpan` units, which are in terms of `100ns` ticks.

**Pull request merged**